### PR TITLE
Serialize Json direct to Http PipeWriter

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Serialize Json direct to Kestrel's Http `PipeWriter` rather than intermediary buffer and then flush
- Switch off `AllowSynchronousIO` as no longer needed

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [x] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Runner/JsonRpc/Startup.cs [hunk 1]
```diff
@@ -166,20 +165,20 @@ ValueTask<long> SerializeTimeoutException(IJsonRpcService service, Stream result
                     try
                     {
                         JsonRpcContext jsonRpcContext = JsonRpcContext.Http(jsonRpcUrl);
+                        long totalResponseSize = 0;
                         await foreach (JsonRpcResult result in jsonRpcProcessor.ProcessAsync(request, jsonRpcContext))
                         {
-                            Stream resultStream = jsonRpcConfig.BufferResponses ? new MemoryStream() : ctx.Response.Body;
+                            Stream stream = jsonRpcConfig.BufferResponses ? new MemoryStream() : null;
+                            ICountingBufferWriter resultWriter = stream is not null ? new CountingStreamPipeWriter(stream) : new CountingPipeWriter(ctx.Response.BodyWriter);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Runner/JsonRpc/Startup.cs:172]: You can potentially merge this into one, but you access stream later

**Review by LukaszRozmej — COMMENTED** [review]
Can you explain more reasoning about changes? Or do you have some numbers?

**Comment by benaadams** [comment]
> Can you explain more reasoning about changes? Or do you have some numbers?

Currently it writes to an intermediate PipeWriter which then periodically writes to the HttpSteam which then writes to the Http PipeWriter

Change skips the intermediate PipeWriter, and the HttpStream by plugging directly into Kestrel's Http PipeWriter so reduces a lot of abstraction redirections and also several memory copies and buffering. Raw as you can go.

File: src/Nethermind/Nethermind.Runner/JsonRpc/Startup.cs (hunk 1 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Runner/JsonRpc/Startup.cs:172]: Need the stream to write later when buffered

**Comment by LukaszRozmej** [comment]
ok what about AllowSynchronousIO ?

**Comment by benaadams** [comment]
> ok what about AllowSynchronousIO ?

Don't need it anymore as can synchronously serialize to Kestrel's PipeWriter and it handles all the async behind the scenes; so don't need to allow it. Is essentially all async while appearing to be sync for the Serializer.

**Comment by benaadams** [comment]
Should also reduce latency

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**